frontend/catalyst/compilation_pipelines.py

```diff
@@ -805,6 +816,7 @@ def qjit(
     verbose=False,
     logfile=None,
     pipelines=None,
+    abstracted_axes=None,
```

Comment:
It does work well with AOT mode. That is the reason behind https://github.com/google/jax/pull/18579. The user can still use `jax.core.ShapedArray([3], float)` but have `abstracted_axes={0: "n"}` and it works.

Reply:
Nice! Still strange though that you would need to specify a concrete shape, even if you want it to be dynamic.

Reply:
I think this comes from actual concrete arrays. Imagine there was no option for AOT compile and only compile once you have the concrete arguments to the function. The concrete arguments is a concrete array with a concrete shape. In other words:

```python
def foo(x, abstracted_axes={0 : "n"}):
  ...
  
foo([1, 1, 1])
```

should still work.

Reply:
Agreed, but I think that's another reason we might want to consider our own UI for this feature :)

Reply:
A note from my side regarding the abstracted axis parameter: I think we need to investigate (and test) this mechanism more because I believe that its main purpose is to declare that different axes should be described by a shared variable. The conversion from static to dynamic shape seems to me a smaller task for it. Not sure if we must do this investigation in this PR though.

Reply:
There is a test where multiple arguments share the same parameter. You can already see the JAXPR and the MLIR is correctly generated. And this works for ShapedArray because we convert the concrete arguments to ShapedArray before compiling.

Reply:
@erick-xanadu yes, I can see these tests, thanks. I also added a lit-test making more than one axes dynamic.

Reply:
@josh146 will this feature / improving the UI be in scope for this PR or be another PR? I'll leave this open for the time being.

Reply:
A new/simplified UI to specify abstracted axes is out of scope for this story/epic, I would say :) Something we can add to the plan to explore, but I'm hesitant to design a new UI without getting a sense of how the current one works in practice.

Reply:
Alright, I'm resolving this issue then! Thanks for clarifying that @josh146 . We can continue the discussion of how `abstracted_axes` affects users in the other tickets / or via slack.